relentless/variable.py

```diff
@@ -305,6 +305,65 @@ def athigh(self):
         """
         return self.state is self.State.HIGH
 
+    def __add__(self, val):
```

Comment:
These operations are actually more generic than just the `DesignVariables`. Hence, I recommend moving the methods as follows:

I think the binary operators (`__add__`, etc.) should be defined at the `Variable` level. If a value is not a `Variable`, you should coerce it into an `IndependentVariable`.

I think the unary infix operators (`__iadd__`, etc.) should be defined at the `IndependentVariable` level, and they should only allow regular numbers (and not `Variable`), as this would otherwise mean that we need to convert the variable into a `DependentVariable` However, infix operators may not change the type of the object itself. I think it is reasonable to scope this separation.